docs/proposals/20210310-opt-in-autoscaling-from-zero.md

```diff
@@ -0,0 +1,242 @@
+---
+title: Opt-in Autoscaling from Zero
+authors:
+  - "@elmiko"
+reviewers:
+  - "@"
+creation-date: 2021-03-10
+last-updated: 2021-03-10
+status: implementable
+---
+
+# Opt-in Autoscaling from Zero
+
+## Table of Contents
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+    - [Security Model](#security-model)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan](#test-plan-optional)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+* **Node Group** This term has special meaning within the cluster autoscaler, it refers to collections
+  of nodes, and related physical hardware, that are organized within the autoscaler for scaling operations.
+  These node groups do not have a direct relation to specific CRDs within Kubernetes, and may be handled
+  differently by each autoscaler cloud implementation. In the case of Cluster API, node groups correspond
+  directly to MachineSets and MachineDeployments that are marked for autoscaling.
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+## Summary
+
+The [Kubernetes cluster autoscaler](https://github.com/kubernetes/autoscaler) currently supports
+scaling on Cluster API deployed clusters for MachineSets and MachineDeployments. One feature
+that is missing from this integration is the ability to scale down to, and up from, a MachineSet
+or MachineDeployment with zero replicas.
+
+This proposal defines an opt-in mechanism whereby Cluster API users and cloud providers can define
```

Comment:
I woud suggest to use infrastructure providers instead of cloud providers, given that AFAIK this proposal targets all the CAPI infrastructure providers, not only the ones running on a real cloud infrastructure.
```suggestion
This proposal defines an opt-in mechanism whereby Cluster API users and infrastructure providers can define
```
Same down in the document

Reply:
good suggestion, i will change those references